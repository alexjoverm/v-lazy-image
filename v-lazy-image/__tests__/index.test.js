import { mount } from "@vue/test-utils";
import Component from "../index";

describe("v-lazy-image", () => {
  beforeEach(() => {
    window.IntersectionObserver = jest.fn(() => ({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    }));
  });

  test("plain img can be created", () => {
    const wrapper = mount(Component, {
      propsData: {
        src: "http://lorempixel.com/400/200/",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
  test("using picture can be created", () => {
    const wrapper = mount(Component, {
      propsData: {
        src: "http://lorempixel.com/400/200/",
        usePicture: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});

// test("picture renders correctly", () => {
//   let intersect;
//   global.IntersectionObserver = function (cb) {
//     intersect = cb;
//     return {
//       observe: () => {},
//       disconnect: () => {},
//     };
//   };
//   let source = '<source srcset="test" />';
//   const wrapper = shallowMount(Component, {
//     slots: {
//       default: [source, source],
//     },
//     propsData: {
//       src: "http://lorempixel.com/400/200/",
//       srcset: "image-320w.jpg 320w, image-480w.jpg 480w",
//       sizes: "(max-width: 320px) 280px, 440px",
//       alt: "Test",
//       usePicture: true,
//     },
//   });
//   intersect([{ isIntersecting: true }]);
//   expect(wrapper.element).toMatchSnapshot();
// });

// test("picture element lazy loads", () => {
//   let intersect;
//   global.IntersectionObserver = function (cb) {
//     intersect = cb;
//     return {
//       observe: () => {},
//       disconnect: () => {},
//     };
//   };
//   let source = '<source srcset="test" />';
//   let src = "http://lorempixel.com/400/200/";
//   const wrapper = shallowMount(Component, {
//     slots: {
//       default: [source, source],
//     },
//     propsData: {
//       src,
//       usePicture: true,
//     },
//   });
//   expect(wrapper.vm.intersected).toBe(false);
//   expect(wrapper.isEmpty()).toBe(true);
//   intersect([{ isIntersecting: true }]);
//   let imgWrapper = wrapper.find("img");
//   expect(imgWrapper.is("img")).toBe(true);
//   expect(wrapper.vm.intersected).toBe(true);
//   expect(imgWrapper.attributes("src")).toBe(src);
// });
