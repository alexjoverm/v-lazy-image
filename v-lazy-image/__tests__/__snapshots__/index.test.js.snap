// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v-lazy-image plain img can be created 1`] = `
<img
  class="v-lazy-image"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
/>
`;

exports[`v-lazy-image using picture can be created 1`] = `
<picture>
  <img
    class="v-lazy-image"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs="
  />
</picture>
`;
